beets/ui/__init__.py

```diff
@@ -559,6 +673,16 @@ def _colordiff(a, b, highlight='text_highlight',
     a_out = []
     b_out = []
 
+    add_mapper = (
+        lambda w: w if re.match(r"(\s)", w) else colorize(highlight_added, w)
+    )
+    remove_mapper = (
+        lambda w: w if re.match(r"(\s)", w) else colorize(highlight_removed, w)
+    )
+    minor_mapper = (
+        lambda w: w if re.match(r"(\s)", w) else colorize(minor_highlight, w)
+    )
```

Comment:
These could use [`w.isspace()`](https://docs.python.org/3.10/library/stdtypes.html#str.isspace) instead of resorting to full-blown regexes.

<del>Additionally, wouldn't it make sense to move the check for empty or whitespace-only strings into `colorize` itself, and get rid of the `*_mapper`s completely?</del> Nevermind, I misunderstood the purpose of having these mappers (only) in this function. However, see below for another code suggestion.

Reply:
I'm re-reviewing this myself (it was inherited from the previous maintainer of this PR) and I agree that these regexes are confusing and/or unnecessary. At one point I think I understood why these mappers existed but now I've arrived at your conclusion - why can't we check for empty whitespace elsewhere? I notice you have an answer but would appreciate it if you could share! 

In the meantime I did try your suggestion below and unfortunately re.Match objects aren't iterable, although the .isspace() option will simplify this somewhat anyway.

Reply:
> At one point I think I understood why these mappers existed but now I've arrived at your conclusion - why can't we check for empty whitespace elsewhere? I notice you have an answer but would appreciate it if you could share!

I believe the point is that it's not always clear whether we want to "highlight" whitespace (note that the background color might be different depending on whether colorization is applied or not. That's generally a design decision that may depend on context. Here, whoever implemented this first chose to not highlight whitespace (changes). But not every caller of `colorize` might want that behaviour.

> In the meantime I did try your suggestion below and unfortunately re.Match objects aren't iterable, although the .isspace() option will simplify this somewhat anyway.

What exactly did you try? There's a small error in my suggestion below, it should have been

```python
b_out.append(re.sub(r"\S+", lambda s: colorize(s, highlight_added), b))
```
note the added argument `b`.

Alternatively, I just thought of a different way to achive the conditional whitespace highlighting: Add a keyword argument to `colorize`, as in `def colorize(color_name, text, ws_highlight=True)` and handle the `ws_highlight=False` case in `colorize`.

Reply:
I've had another look at the `re` suggestion, and I see what you meant when saying that `re.Match` objects aren't iterable. It's easy to solve though:
```
b_out.append(re.sub(r"\S+", lambda s: colorize(s.group(0), highlight_added), b))
```
cf. the example here: https://docs.python.org/3.10/library/stdtypes.html#str.title

Reply:
I've simplified this by implementing your suggestion of moving whitespace highlighting to colorize :)